data-structures/unixfsv2.md

```diff
@@ -0,0 +1,68 @@
+# UnixFSv2 Schema
+
+This doc contains the File and Directory schemas for UnixFSv2.
+
+This schema depends on some Advanced Layouts and types defined
+in other schemas:
+
+* [Data](https://github.com/ipld/specs/pull/211)
+* [Hashmap](https://github.com/ipld/specs/blob/master/data-structures/hashmap.md)
+
+```sh
+type DirData union {
+  | EntryMap "map"
+  | Hamt "hamt"
```

Comment:
root node becomes inlined here, maybe this should be `&Hamt`? (also still hate using the specific algorithm name but know I'm alone on this so won't push).

Reply:
isn’t the root node usually pretty small? why wouldn’t we want to inline it?

Reply:
root node will end up being about the same size as the intermediate node and could be roughly in line with the average leaf node size depending on various factors. The root node is just a special intermediate node that's at depth=0.

Reply:
ya, i think we should just inline it then. the rest of the directory metadata is so minimal that it doesn’t make a ton of sense to have it in a separate block.